### PR TITLE
feat(cubestore): CC-1133 - add headers to telemetry request

### PR DESCRIPTION
**Pull request by wolfeeeg**
**Check List**
- [ ] Tests has been run in packages where changes made if available
- [ ] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

[For example #12]

**Description of Changes Made (if issue reference is not provided)**

Add headers to telemetry request

### CODE REVIEW

File: rust/cubestore/cubestore/src/telemetry/mod.rs [hunk 1]
```diff
@@ -74,9 +77,16 @@ impl TelemetryTransport for HttpTelemetryTransport {
...
                 .post(&self.endpoint_url)
+                .headers(
+                    self.headers
+                        .as_ref()
+                        .unwrap_or(&empty_header_map)
+                        .to_owned(),
+                )
                 .json(&to_send)
                 .send()
                 .await?;
```
Comment (waralexrom) [inline rust/cubestore/cubestore/src/telemetry/mod.rs:92]:
```suggestion
            let header_map = self
                .headers
                .as_ref()
                .map_or_else(|| HeaderMap::new(), |m| m.to_owned());
            let res = self
                .client
                .post(&self.endpoint_url)
                .headers(header_map)
                .json(&to_send)
                .send()
                .await?;

```

Comment (waralexrom) [reply rust/cubestore/cubestore/src/telemetry/mod.rs:92]: The motivation is not to create an empty `HeaderMap` if we don't need it.

**Review by waralexrom — COMMENTED** [review]
We also need a simple test here

**Comment by wolfeeeg** [comment]
> We also need a simple test here

added

**Review by waralexrom — APPROVED** [review]
LGTM :+1:

**Pull request merged**